### PR TITLE
Exclude Helm tests from dry-run diff

### LINKED ISSUE #131: Exclude Helm tests from dry-run diff
Currently, [Helm tests](https://helm.sh/docs/topics/chart_tests/) are rendered in the diff as additions. They should be ignored instead

### PR DESCRIPTION
**Pull request by raminqaf**
Fixes #131

### CODE REVIEW

**Review by disrupted — CHANGES_REQUESTED** [review]

File: tests/components/test_kubernetes_app.py [hunk 1]
```diff
@@ -69,6 +69,7 @@ def test_should_lazy_load_helm_wrapper_and_not_repo_add(
         mocker: MockerFixture,
         helm_mock: MagicMock,
         app_value: KubernetesTestValue,
+        dry_run_handler: MagicMock,
```
[the remark this answers is not in the document]
Comment (raminqaf) [reply tests/components/test_kubernetes_app.py:72]: It is needed in this case because the dry-run is set to try when calling the `deploy`, and the dry-run handler is executed. When passing the MagicMock, the actual execution is skipped. I can add a validation `assert_called_once`.
Another way would be to remove it and call deploy with dry-run Flase.

File: tests/component_handlers/helm_wrapper/test_helm_wrapper.py [hunk 2]
```diff
@@ -320,9 +320,10 @@ def test_raise_parse_error_when_helm_content_is_invalid(self):
         stdout = """---
             # Resource: chart/templates/test1.yaml
             """
-        with pytest.raises(ParseError) as value_error:
-            list(Helm.load_manifest(stdout))
-        assert str(value_error.value) == f"Failed to parse Helm stdout:\n {stdout}"
+        with pytest.raises(ParseError) as parse_error:
+            helm_template = list(Helm.load_manifest(stdout))
+            assert len(helm_template) == 0
+            assert str(parse_error.value) == "Not a valid Helm template source"
```
Comment (disrupted) [inline tests/component_handlers/helm_wrapper/test_helm_wrapper.py:326]: assertion needs to be outside the context manager scope. this is a common mistake we used to make throughout the tests. see https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

Comment (disrupted) [reply tests/component_handlers/helm_wrapper/test_helm_wrapper.py:326]: or you could rewrite it as
```python
with raises(ParseError, match="Not a valid Helm template source"):
    ...
```

File: tests/component_handlers/helm_wrapper/test_helm_wrapper.py [hunk 3]
```diff
@@ -362,14 +377,14 @@ def test_load_manifest(self):
             isinstance(helm_template, HelmTemplate) for helm_template in helm_templates
         )
         assert helm_templates[0].filepath == "chart/templates/test3a.yaml"
-        assert helm_templates[0].template == {"data": [{"a": 1}, {"b": 2}]}
```
Comment (disrupted) [inline tests/component_handlers/helm_wrapper/test_helm_wrapper.py:365]: why did you delete this line?

File: tests/components/test_kubernetes_app.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (raminqaf) [reply tests/components/test_kubernetes_app.py:72]: IMO the correct way would be to split this into two tests. Once with dry_run true and once with false. But this is outside of the scope of this PR. So maybe setting it to false is the better approach for now

File: tests/component_handlers/helm_wrapper/test_helm_wrapper.py (hunk 3 above)
Comment (raminqaf) [reply tests/component_handlers/helm_wrapper/test_helm_wrapper.py:365]: ups

File: tests/components/test_kubernetes_app.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (raminqaf) [reply tests/components/test_kubernetes_app.py]: Changed

File: tests/component_handlers/helm_wrapper/test_helm_wrapper.py [hunk 4]
```diff
@@ -362,14 +377,14 @@ def test_load_manifest(self):
...
         assert helm_templates[1].filepath == "chart/templates/test3b.yaml"
         assert helm_templates[1].template == {"foo": "bar"}
 
     def test_get_manifest(self, run_command: MagicMock, mock_get_version: MagicMock):
         helm_wrapper = Helm(helm_config=HelmConfig())
         run_command.return_value = """Release "test-release" has been upgraded. Happy Helming!
 NAME: test-release
+MANIFEST:
```
Comment (disrupted) [inline tests/component_handlers/helm_wrapper/test_helm_wrapper.py:387]: why did you add this line, is it guaranteed that the Helm template always includes this line?
otherwise, I'd say we should leave the test as-is.

Comment (raminqaf) [reply tests/component_handlers/helm_wrapper/test_helm_wrapper.py:387]: I added this because this message represents the stdout of `helm upgrade --install,` and this command outputs these headers, HOOKS, MANIFEST, and NOTES (optional).
On the other hand, the command `helm get manifest` only prints the manifests without any header. That is why I added the check if the `MANIFEST:` header exists, otherwise the start should be `---`

Comment (raminqaf) [reply tests/component_handlers/helm_wrapper/test_helm_wrapper.py:387]: Actually this is the output of `helm get manifest` I'll correct the string :D

Comment (disrupted) [reply tests/component_handlers/helm_wrapper/test_helm_wrapper.py]: I would say revert this line or even remove the whole header if that's the output format of `helm get manifest`

File: tests/component_handlers/helm_wrapper/test_helm_wrapper.py [hunk 5]
```diff
@@ -278,14 +279,17 @@ def test_validate_console_output(self):
...
     def test_helm_template_load(self):
-        stdout = """---
-# Source: chart/templates/test2.yaml
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-    labels:
-        foo: bar
-"""
+        stdout = dedent(
```
[the remark this answers is not in the document]
Comment (raminqaf) [reply tests/component_handlers/helm_wrapper/test_helm_wrapper.py:282]: These disrupted multi-line strings always annoyed me! But now it is clean 🧼

File: tests/component_handlers/helm_wrapper/test_helm_wrapper.py [hunk 6]
```diff
@@ -331,6 +316,16 @@ def test_load_manifest(self):
...
+    def test_raise_parse_error_when_helm_content_is_invalid(self):
+        stdout = """---
+            # Resource: chart/templates/test1.yaml
+            """
+        with pytest.raises(ParseError, match="Not a valid Helm template source"):
+            helm_template = list(Helm.load_manifest(stdout))
+            assert len(helm_template) == 0
+
+    def test_load_manifest(self):
+
```
[the remark this answers is not in the document]
Comment (disrupted) [reply tests/component_handlers/helm_wrapper/test_helm_wrapper.py]: #294
